### PR TITLE
ModAssignmentDrawer: show energy changes to make mods fit with fewest materials spent

### LINKED ISSUE #7247: LO-generated loadout can't slot all mods
I generated this loadout by selecting these mods and choosing the top loadout in LO. However when I go to the loadout itself and ask for mod assignments, it now says it can't slot two of the mods: 
Screen Shot 2021-11-11 at 11 29 17 AM

### LINKED ISSUE #9017: Show mod placement should also show energy change hints
It's not that helpful to know my mods can't fit. Tell me how to fit them!

We should also show the costs for any upgrades.

### LINKED ISSUE #9162: LO choose class item
This may be happening implicitly now, but we should make sure to include tags (favorite, etc) in the criteria for which class item to choose

### PR DESCRIPTION
**Pull request by robojumper**
grafik

Fixes #7247.
Fixes #9017.
Fixes #9162.

### CODE REVIEW

**Comment by bhollis** [comment]
Very excited about this!! I'll check out the code tomorrow.

**Comment by bhollis** [comment]
Two possible UX notes:

1. It'd be great to have a tooltip for the upgrade - the "X -> Y" notation has confused several people.
2. It is probably worth explaining the Upgrade Costs and Unassigned mods a bit more. e.g. "X mods didn't fit, you need to perform the upgrades shown below each item which will cost Y in total"

**Comment by robojumper** [comment]
Added some tooltips and descriptions, and unified more logic and display between LO set row and mod assignment drawer. nev might have opinions on the particular wording.

**Comment by nev-r** [comment]
ok so this will sound crazy and picky, but:
the implication of saying "there's a problem and here's what it will cost to fix it" _then_ saying "one mod doesn't fit",
sort of sounds like it's saying that **even after the upgrades**, one won't fit

so preferably we say "here's the mod we are spending, to make fit"
suggestions incoming shortly, just want to note this

**Comment by nev-r** [comment]
ok then instead that will need to be more explicit like "these mods still won't fit after upgrades" otherwise that last section looks like it's a list of the ones affected by the above problem

File: config/i18n.json [hunk 1]
```diff
@@ -654,7 +655,17 @@
...
     "MaximizeStat": "Maximize Stat",
-    "ModPlacement": "Mod Placement",
+    "ModPlacement": {
+      "InvalidMods": "Invalid Mods",
+      "InvalidModsDesc": "1 mod cannot fit into any item.",
+      "InvalidModsDesc_plural": "{{count}} mods cannot fit into any item.",
+      "ModPlacement": "Mod Placement",
+      "UnassignedMods": "Unassigned Mods",
+      "UnassignedModsDesc": "1 mod did not fit due to insufficient energy capacity or mod slots.",
+      "UnassignedModsDesc_plural": "{{count}} mods did not fit due to insufficient energy capacity or mod slots.",
```
Comment (nev-r) [inline config/i18n.json:665]:
```suggestion
      "UnassignedModsDesc": "1 mod did not fit due to insufficient energy capacity or mod slots. Energy upgrades to the selected armor will not fix the issue.",
      "UnassignedModsDesc_plural": "{{count}} mods did not fit due to insufficient energy capacity or mod slots. Energy upgrades to the selected armor will not fix the issue.",
```
maybe this? struggling to make it modular enough but this seems exhaustively true

**Review by bhollis — APPROVED** [review]
Code looks great, functionality works well.

**Pull request merged**